### DISCUSSION TITLE
Can we customize Material Toolkit components, such as text fields, chips, and cards, to suit our requirements?

### DISCUSSION THREAD

**Discussion by Mamtasawant889** [unoplatform/uno#19039] [Q&A]
Can we customize Material Toolkit components, such as text fields, chips, and cards, to suit our requirements?

**Comment by morning4coffe-dev** [comment]
Hey @Mamtasawant889,

Thank you for your question! **TL;DR** answer is: YES, you can!  

If you don't need much customization, you can use the [Material Design System Properties (DSP) override](https://platform.uno/docs/articles/external/uno.themes/doc/material-dsp.html), which allows you to easily customize the colors across your app.  

For further customization, you can:
- Use [Lightweight styling](https://platform.uno/docs/articles/external/uno.themes/doc/lightweight-styling.html) for any WinUI or Toolkit components.
- Use the same "full" [styling methods](https://learn.microsoft.com/en-us/windows/apps/develop/platform/xaml/xaml-styles) as for other WinUI components.

You can find the reference for all the Toolkit controls [here](https://platform.uno/docs/articles/external/uno.toolkit.ui/doc/controls/AutoLayoutControl.html). Let me know, if this helps.

**Reply by DevTKSS** [reply]
Reading the question asked here again I am not sure @Mamtasawant889 is meaning the colors thats modifiable with the dsp. Maybe more like a... If you would want to take a xy ui element which is brought by the toolkit and add lets say a text or button or whatever like doing CustomControl/UserControl i would understand this question.
And i would agree, that this explainations for uno apps (there are some specifics like handling of bindings with the utu: bindings i am also struggling with. So maybe here is wanted to know how to do such things.
@Mamtasawant889 could you point out your question a bit more?
